driver-core/src/main/com/mongodb/internal/TimeoutContext.java

```diff
@@ -265,23 +270,53 @@ public static Timeout calculateTimeout(@Nullable final Long timeoutMS) {
         return null;
     }
 
+    /**
+     * Returns the computed server selection timeout
+     *
+     * <p>Caches the computed server selection timeout if:
+     * <ul>
+     *     <li>not in a maintenance context</li>
+     *     <li>there is a timeoutMS, so to keep the same legacy behavior.</li>
+     *     <li>the server selection timeout is less than the remaining overall timeout.</li>
+     * </ul>
+     *
+     * @return the timeout context
+     */
     public Timeout computedServerSelectionTimeout() {
```

Comment:
[optional] How about renaming the method to `computeServerSelectionTimeout` for consistency with verb-first naming in this class (i.e as 'start')?